### PR TITLE
Fingerbank integration

### PR DESCRIPTION
**Pull request by julsemaan**
Was reverted so it can be reviewed quickly

### CODE REVIEW

File: db/upgrade-X.X.X-X.Y.Z.sql [hunk 1]
```diff
@@ -91,3 +91,10 @@ INSERT INTO `person` (pid,notes) VALUES ("default","Default User - do not delete
 --
 
 UPDATE `node` SET pid = 'default' WHERE status = 'unreg' AND pid = 'admin';
+
```
Comment (julsemaan) [inline db/upgrade-X.X.X-X.Y.Z.sql:94]: Should remove the os table

File: html/pfappserver/lib/pfappserver/Form/Config/Fingerbank/Device.pm [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+=cut
+
+use HTML::FormHandler::Moose;
+extends 'pfappserver::Base::Form';
+with 'pfappserver::Base::Form::Role::Help';
+
+use pf::admin_roles;
+use pf::log;
+
+has roles => ( is => 'rw', default => sub { [] } );
```
Comment (julsemaan) [inline html/pfappserver/lib/pfappserver/Form/Config/Fingerbank/Device.pm:20]: why roles ?

File: html/pfappserver/lib/pfappserver/Form/Config/Fingerbank/DHCP_Vendor.pm [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+=cut
+
+use HTML::FormHandler::Moose;
+extends 'pfappserver::Base::Form';
+with 'pfappserver::Base::Form::Role::Help';
+
+use pf::admin_roles;
+use pf::log;
+
+has roles => ( is => 'rw', default => sub { [] } );
```
Comment (julsemaan) [inline html/pfappserver/lib/pfappserver/Form/Config/Fingerbank/DHCP_Vendor.pm:20]: why roles ?

File: html/pfappserver/lib/pfappserver/Form/Config/Fingerbank/DHCP_Fingerprint.pm [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+=cut
+
+use HTML::FormHandler::Moose;
+extends 'pfappserver::Base::Form';
+with 'pfappserver::Base::Form::Role::Help';
+
+use pf::admin_roles;
+use pf::log;
+
+has roles => ( is => 'rw', default => sub { [] } );
```
Comment (julsemaan) [inline html/pfappserver/lib/pfappserver/Form/Config/Fingerbank/DHCP_Fingerprint.pm:20]: why roels ?

File: html/pfappserver/lib/pfappserver/Form/Config/Fingerbank/MAC_Vendor.pm [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+=cut
+
+use HTML::FormHandler::Moose;
+extends 'pfappserver::Base::Form';
+with 'pfappserver::Base::Form::Role::Help';
+
+use pf::admin_roles;
+use pf::log;
+
+has roles => ( is => 'rw', default => sub { [] } );
```
Comment (julsemaan) [inline html/pfappserver/lib/pfappserver/Form/Config/Fingerbank/MAC_Vendor.pm:20]: why roles ?

File: html/pfappserver/lib/pfappserver/Form/Config/Fingerbank/Settings.pm [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+        push @$list, $section => {id => $section, type => 'Compound'};
+        foreach my $parameter ( keys %{$config->{$section}} ) {
+            my $field_name = $section . "." . $parameter;
+            my $field_doc = $config_doc->{$field_name};
+            $field_doc->{description} =~ s/\n//sg;
+
+            my $field = {
+                id      => $field_name,
+                label   => $field_name,
+#                element_attr => { 'placeholder' => $config_defaults->{$field_name} },
```
Comment (julsemaan) [inline html/pfappserver/lib/pfappserver/Form/Config/Fingerbank/Settings.pm:41]: remove if useless

File: html/pfappserver/lib/pfappserver/Form/Config/Fingerbank/User_Agent.pm [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+=cut
+
+use HTML::FormHandler::Moose;
+extends 'pfappserver::Base::Form';
+with 'pfappserver::Base::Form::Role::Help';
+
+use pf::admin_roles;
+use pf::log;
+
+has roles => ( is => 'rw', default => sub { [] } );
```
Comment (julsemaan) [inline html/pfappserver/lib/pfappserver/Form/Config/Fingerbank/User_Agent.pm:20]: why roles ?

File: html/pfappserver/lib/pfappserver/PacketFence/Controller/Config/Fingerbank/DB.pm [hunk 8]
```diff
@@ -0,0 +1,83 @@
+package pfappserver::PacketFence::Controller::Config::Fingerbank::DB;
```
Comment (julsemaan) [inline html/pfappserver/lib/pfappserver/PacketFence/Controller/Config/Fingerbank/DB.pm:1]: I can't believe I'm already this far and I didn't see a map grep sort uniq map

File: html/pfappserver/lib/pfappserver/PacketFence/Controller/Config/Fingerbank/Settings.pm [hunk 9]
```diff
@@ -0,0 +1,150 @@
...
+
+=cut
+
+sub check_for_api_key :Private {
+    my ( $self, $c ) = @_;
+    my $logger = pf::log::get_logger;
+
+    if ( !fingerbank::Config::is_api_key_configured ) {
+        $logger->warn("Fingerbank API key is not configured. Running with limited features");
+        my $status_msg = "It looks like Fingerbank API key is not configured. You may have forgot the onboard process. To fully beneficiate of Fingerbank, please proceed here: https://fingerbank.inverse.ca/onboard";
```
Comment (julsemaan) [inline html/pfappserver/lib/pfappserver/PacketFence/Controller/Config/Fingerbank/Settings.pm:32]: perhaps you forgot to load fingerbank ?

File: html/pfappserver/root/static/admin/config/fingerbank-devices.js [hunk 10]
```diff
@@ -0,0 +1,72 @@
+$(function() { // DOM ready
+    var items = new FingerBankDevices();
```
Comment (julsemaan) [inline html/pfappserver/root/static/admin/config/fingerbank-devices.js:2]: Fingerbank not FingerBank

File: html/pfappserver/root/static/admin/config/violations.js [hunk 11]
```diff
@@ -1,5 +1,29 @@
...
 
+    function getTrigger(query, process) {
+        var input = $('#violationTriggers [data-provide="typeahead"]');
+        var trigger_type = $('#trigger_type');
+        var type = trigger_type.val();
+        //console.log(trigger_type);
+        var control = input.closest('.control-group');
+        $.ajax('/trigger/search/' + type + "/" + query)
+            .done(function(data) {
+                /*
```
Comment (julsemaan) [inline html/pfappserver/root/static/admin/config/violations.js:11]: remove if useless

File: lib/pf/triggerParser.pm [hunk 12]
```diff
@@ -0,0 +1,79 @@
+package pf::triggerParser;
+=head1 NAME
+
+pf::triggerParser - Trigger for openvas
```
Comment (julsemaan) [inline lib/pf/triggerParser.pm:4]: doesn't look like openvas to me

File: html/pfappserver/lib/pfappserver/Form/Config/Fingerbank/Settings.pm (hunk 6 above)
Comment (dwlfrth) [inline html/pfappserver/lib/pfappserver/Form/Config/Fingerbank/Settings.pm:41]: Keep it for the moment. It is to integrate the defaults values...

**Pull request closed without merging**